### ISSUE TITLE
what is the parameter for provisioning a enhanced OKE cluster rather than type Basic 

### ISSUE THREAD

**Issue by csraovaradev** [oracle/terraform-provider-oci#1820]
This Github template is intended for questions regarding the Terraform **Oracle Cloud Infrastructure** provider.

Before using this template, check whether your question is already answered in one of the following guides:
* [Frequently Asked Questions](https://docs.oracle.com/en-us/iaas/Content/API/SDKDocs/terraform.htm)
* [Troubleshooting](https://docs.oracle.com/en-us/iaas/Content/API/SDKDocs/terraformtroubleshooting.htm)

If you have a support request related to Oracle Cloud Infrastructure services, please [contact Oracle support](https://docs.cloud.oracle.com/iaas/Content/GSG/Tasks/contactingsupport.htm)

If you have a support request or question related to core Terraform functionality, please submit them to one of these resources:

* [Terraform community resources](https://www.terraform.io/docs/extend/community/index.html)
* [HashiCorp support](https://support.hashicorp.com) (Terraform Enterprise customers)

**Comment by JanisOrlovs** [comment]
Hello,
For me it's also good question. There is no option to set neither basic, neither enachanced. 

Plus there is no way how to codify upgrade from Basic to Enchanced

**Comment by etyoma** [comment]
Yes, this was promised to be released within of a week  two or three weeks ago.
Probably it, was, but is still not reflected anyhow in the documentation.

**Comment by teresa-magicbeans** [comment]
I just used @etyoma example and it worked, i just added: 
type               = "ENHANCED_CLUSTER"
to my oci_containerengine_cluster resource, and it worked. 

resource "oci_containerengine_cluster" "test_cluster" {
    #Required
    compartment_id     = var.compartment_ocid
    kubernetes_version = var.kubernetes_version
    name               = "tfTestCluster"
    vcn_id             = oci_core_vcn.test_vcn.id
    type               = "ENHANCED_CLUSTER"
}

**Comment by tf-oci-pub** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by tf-oci-pub** [comment]
Thank you for reporting the issue. We observed the affected resources are not provided in the description or it's incorrect. We request you to add it in issue description as mentioned in below format.
Example: affected_resources = oci_core_instance , oci_core_instances
 
If it's not related to any particular resource then mention affected resource as terraform.
Example: affected_resources = terraform
 
As this works through automation, request you to follow exact syntax.
